pkg/cstorpool/v1alpha2/cstorpool_test.go

```diff
@@ -168,3 +163,116 @@ func TestCstorPoolList(t *testing.T) {
+		"UID set 8":  {[]string{"uid1", "uid2", "uid3", "uid4", "uid5", "uid6", "uid7"}},
+		"UID set 9":  {[]string{"uid1", "uid2", "uid3", "uid4", "uid5", "uid6", "uid7", "uid8"}},
+		"UID set 10": {[]string{"uid1", "uid2", "uid3", "uid4", "uid5", "uid6", "uid7", "uid8", "uid9"}},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			lb := ListBuilder().WithUIDs(mock.expectedUIDs...).List()
+			output := ListBuilder().WithList(lb).List().GetPoolUIDs()
+			if len(lb.GetPoolUIDs()) != len(output) {
+				t.Fatalf("Test %v failed, Expected %v Got %v", name, lb.GetPoolUIDs(), output)
+			}
+
+		})
+	}
+}
+
+func TestNewListFromUIDNode(t *testing.T) {
+	tests := map[string]struct {
+		UIDNodeMap    map[string]string
+		expectedPools []string
+	}{
+		"Test 1": {map[string]string{"Pool 1": "host 1"}, []string{"Pool 1"}},
+		"Test 2": {map[string]string{"Pool 1": "host 1", "Pool 2": "host 2"}, []string{"Pool 1", "Pool 2"}},
+		"Test 3": {map[string]string{"Pool 1": "host 1", "Pool 2": "host 2", "Pool 3": "host 3"}, []string{"Pool 1", "Pool 2", "Pool 3"}},
+		"Test 4": {map[string]string{"Pool 1": "host 1", "Pool 2": "host 2", "Pool 3": "host 3", "Pool 4": "host 4"}, []string{"Pool 1", "Pool 2", "Pool 3", "Pool 4"}},
+		"Test 5": {map[string]string{"Pool 1": "host 1", "Pool 2": "host 2", "Pool 3": "host 3", "Pool 4": "host 4", "Pool 5": "host 5"}, []string{"Pool 1", "Pool 2", "Pool 3", "Pool 4", "Pool 5"}},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			output := newListFromUIDNode(mock.UIDNodeMap).GetPoolUIDs()
+			if len(output) != len(mock.expectedPools) {
+				t.Fatalf("Test %v failed: Expected %v but got %v", name, mock.expectedPools, output)
+			}
+
+		})
+	}
+}
+
+func TestNewListFromUIDs(t *testing.T) {
+	tests := map[string]struct {
+		PoolUIDs []string
+	}{
+		"Test 1": {[]string{"Pool 1"}},
+		"Test 2": {[]string{"Pool 1", "Pool 2"}},
+		"Test 3": {[]string{"Pool 1", "Pool 2", "Pool 3"}},
+		"Test 4": {[]string{"Pool 1", "Pool 2", "Pool 3", "Pool 4"}},
+		"Test 5": {[]string{"Pool 1", "Pool 2", "Pool 3", "Pool 4", "Pool 5"}},
+	}
+
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			output := newListFromUIDs(mock.PoolUIDs).GetPoolUIDs()
+			if len(output) != len(mock.PoolUIDs) {
+				t.Fatalf("Test %v failed: Expected %v but got %v", name, mock.PoolUIDs, output)
+			}
+
+		})
+	}
+}
+
+func TestTemplateFunctionsCount(t *testing.T) {
+	tests := map[string]struct {
+		expectedLength int
+	}{
+		"Test 1": {2},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			p := TemplateFunctions()
+			if len(p) != test.expectedLength {
+				t.Fatalf("test %q failed: expected items %v but got %v", name, test.expectedLength, len(p))
+			}
+		})
+	}
+}
+
+func TestHasAnnotation(t *testing.T) {
+	tests := map[string]struct {
+		availableAnnotations map[string]string
+	}{
+		"Test 1": {map[string]string{"Anno 1": "Val 1"}},
```

Comment:
Test case data does not reveal what is being tested.

Can we have a bunch of pre-defined annotations for all the test cases.
Each test case can provide a annotation key & corresponding value.
Test case can also have `isFound bool` property.

```go
tests := map[string]struct {
  key string
  value string
  hasAnnotation bool
}{
// ...
}